internal/artifacts/module_arch.go

```diff
@@ -40,29 +41,242 @@ func ExecuteBuild(source, target string, extensions []string, moduleName, platfo
+	wdGetter func() (string, error)) error {
+
+	requiredModules, err := getRequiredModules(mtaObj, module)
+	if err != nil {
+		return err
+	}
+	if len(requiredModules) > 0 {
+		logs.Logger.Infof(dependenciesProcessingMag, module)
+	}
+
+	for _, requiredModule := range requiredModules {
+		moduleLoc, err := getModuleLocation(source, target, requiredModule, extensions, wdGetter)
+		if err != nil {
+			return err
+		}
+		logs.Logger.Infof(buildMsg, requiredModule)
+		err = buildModule(moduleLoc, moduleLoc, requiredModule, "", false, false)
+		if err != nil {
+			return err
+		}
+	}
+
+	if len(requiredModules) > 0 {
+		logs.Logger.Infof(dependenciesProcessingFinishedMag, module)
+	}
+	return nil
+}
+
+func getRequiredModules(mtaObj *mta.MTA, moduleName string) ([]string, error) {
+
+	var requiredModulesNames []string
+	module, err := mtaObj.GetModuleByName(moduleName)
+	if err != nil {
+		return nil, err
+	}
+	for _, requires := range buildops.GetBuildRequires(module) {
+		requiredModule, err := mtaObj.GetModuleByName(requires.Name)
+		if err != nil {
+			return nil, err
+		}
+		requiredForRequiredModule, err := getRequiredModules(mtaObj, requiredModule.Name)
+		if err != nil {
+			return nil, err
+		}
+		for _, reqForRequiredModule := range requiredForRequiredModule {
+			requiredModulesNames = appendIfNotListed(requiredModulesNames, reqForRequiredModule)
+		}
+		requiredModulesNames = appendIfNotListed(requiredModulesNames, requires.Name)
+
+	}
+	return requiredModulesNames, nil
+}
+
+func appendIfNotListed(list []string, element string) []string {
+	exists := false
+	for _, listElement := range list {
+		if listElement == element {
+			exists = true
+			break
+		}
+	}
+	if !exists {
+		list = append(list, element)
+	}
+	return list
+}
+
+func sortSelectedModules(mtaObj *mta.MTA, selectedModules []string) ([]string, error) {
+	selectedModulesMap := make(map[string]bool)
+	for _, selectedModule := range selectedModules {
+		selectedModulesMap[selectedModule] = false
+	}
+	sortedAllModules, err := buildops.GetModulesNames(mtaObj)
+	var allModules []string
+	if err != nil {
+		return nil, err
+	}
+	for _, module := range sortedAllModules {
+		_, selected := selectedModulesMap[module]
+		if selected {
+			allModules = append(allModules, module)
+		}
+	}
+	return allModules, nil
+}
+
+func getModuleLocation(source, target, moduleName string, extensions []string, wdGetter func() (string, error)) (*dir.ModuleLoc, error) {
+	targetDir, err := getSoloModuleBuildAbsTarget(source, target, moduleName, wdGetter)
+	if err != nil {
+		return nil, err
+	}
+
+	loc, err := dir.Location(source, targetDir, dir.Dev, extensions, wdGetter)
+	if err != nil {
+		return nil, err
+	}
+
+	return dir.ModuleLocation(loc), nil
+}
+
+func checkBuildResultsConflicts(mtaObj *mta.MTA, source, target string, extensions []string, modulesNames []string, wdGetter func() (string, error)) error {
+
+	moduleNameResultPathMap := make(map[string]string)
```

Comment:
This can be `map[string]boolean` because you always set the value to empty string. It's only used as a set of the module names.

Reply:
done